### PR TITLE
Implement slash commands

### PR DESCRIPTION
**Pull request by MattBSG**
# Pull request template
Use this template for all contributions made for the repository.

Please indicate all that apply with a checkmark. To make a check, convert brackets below to `[x]`.

* [x] **Feature implementation**
* [x] **Bug fix**
* [ ] **Code change/improvement**
* [x] **String change (i.e. grammar/spelling error or an addition)**

***

## Describe what your pull request does
This PR implements slash commands in Parakarry. Text commands have been entirely superseded by native slash commands in this release. As a side-effect the following function changes are now present:
* Text commands no longer function; slash commands are the new sole interaction method with the bot via Discord
* The flow for creating modmail threads with a user has changed.
    * `!aopen` and `!open` commands have been combined into a single `/open` command
    * `/open` No longer takes a message argument. Instead, it only opens a thread and notifies the user a thread has been opened with them
    * The moderator will need to send their reply via `/areply` or `/reply` after opening a thread in that threads channel 

* Moderators can no longer directly send attachments in the same message as a reply -- this is a limitation with Discord
    * An alternative is upload the image in the thread channel then reply the link to the user

## Related issues and extra details
https://trello.com/c/LzEA1HOT/163-slash-command-components-threads-migration

### CODE REVIEW

**Review by FlapSnapple — DISMISSED** [review]

**Review by mralext20 — APPROVED** [review]

File: bot.py [hunk 1]
```diff
@@ -75,29 +98,57 @@ async def _close(self, ctx, delay: typing.Optional[str]):
...
         await utils._close_thread(self.bot, ctx, self.modLogs)
 
-    @commands.has_any_role(config.modRole)
-    @commands.command(name='reply', aliases=['r'])
-    async def _reply_user(self, ctx, *, content: typing.Optional[str]):
+    @cog_ext.cog_slash(
+        name='reply',
+        guild_ids=guildList,
+        description='Replys to a modmail, with your username',
+        permissions={config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]},
```
Comment (7bitlyrus) [inline bot.py:105]: `{config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]}` should be made into a static variable and passed instead.

File: bot.py [hunk 2]
```diff
@@ -29,19 +31,40 @@
...
-    @commands.command(name='close')
-    async def _close(self, ctx, delay: typing.Optional[str]):
+        loop = bot.loop
+        loop.create_task(slash.sync_all_commands(delete_from_unused_guilds=True))
+
+    @cog_ext.cog_slash(
+        name='close',
+        guild_ids=guildList,
+        description='Closes a modmail thread, optionally with a delay',
+        permissions={config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]},
```
Comment (7bitlyrus) [inline bot.py:56]: `{config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]}` should be made into a static variable and passed instead.

File: bot.py [hunk 3]
```diff
@@ -75,29 +98,57 @@ async def _close(self, ctx, delay: typing.Optional[str]):
...
         await self._reply(ctx, content)
 
-    @commands.has_any_role(config.modRole)
-    @commands.command(name='areply', aliases=['ar'])
-    async def _reply_anon(self, ctx, *, content: typing.Optional[str]):
+    @cog_ext.cog_slash(
+        name='areply',
+        guild_ids=guildList,
+        description='Replys to a modmail, anonymously',
+        permissions={config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]},
```
Comment (7bitlyrus) [inline bot.py:126]: `{config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]}` should be made into a static variable and passed instead.

File: bot.py [hunk 4]
```diff
@@ -152,86 +228,39 @@ async def _reply(self, ctx, content, anonymous=False):
...
-        else:
-            embed.title = '[ANON] Moderator message'
-            embed.set_author(
-                name=f'{ctx.author} ({ctx.author.id}) as r/NintendoSwitch',
-                icon_url='https://cdn.mattbsg.xyz/rns/snoo.png',
+    @cog_ext.cog_slash(
+        name='open',
+        guild_ids=guildList,
+        description='Open a modmail thread with a user, using your username',
+        permissions={config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]},
```
Comment (7bitlyrus) [inline bot.py:242]: `{config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]}` should be made into a static variable and passed instead.

File: bot.py [hunk 5]
```diff
@@ -255,23 +284,33 @@ async def _snippets(self, ctx, *args):
...
-    @commands.has_any_role(config.modRole)
-    @_appeal.command(name='accept')
-    async def _appeal_accept(self, ctx, *, reason):
+    @cog_ext.cog_subcommand(
+        base='appeal',
+        name='accept',
+        guild_ids=guildList,
+        base_description='Make a decision to accept or deny a ban appeal',
+        description='Accept a user\'s ban appeal',
+        base_permissions={config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]},
```
Comment (7bitlyrus) [inline bot.py:293]: `{config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]}` should be made into a static variable and passed instead.

File: bot.py [hunk 6]
```diff
@@ -325,15 +364,35 @@ async def _appeal_accept(self, ctx, *, reason):
...
-    @commands.has_any_role(config.modRole)
-    @_appeal.command(name='deny')
-    async def _appeal_deny(self, ctx, next_attempt, *, reason):
+    @cog_ext.cog_subcommand(
+        base='appeal',
+        name='deny',
+        guild_ids=guildList,
+        base_description='Make a decision to accept or deny a ban appeal',
+        description='Deny a user\'s ban appeal',
+        base_permissions={config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]},
```
Comment (7bitlyrus) [inline bot.py:373]: `{config.guild: [create_permission(config.modRole, SlashCommandPermissionType.ROLE, True)]}` should be made into a static variable and passed instead.

**Review by 7bitlyrus — APPROVED** [review]

**Review by mralext20 — APPROVED** [review]

**Review by FlapSnapple — APPROVED** [review]

**Pull request merged**